packages/insured-bridge-relayer/src/ProfitabilityCalculator.ts

```diff
@@ -0,0 +1,236 @@
+import winston from "winston";
+import Web3 from "web3";
+const { toWei, toBN, fromWei } = Web3.utils;
+const toBNWei = (number: string | number) => toBN(toWei(number.toString()).toString());
+const fixedPoint = toBNWei(1);
+
+import { objectMap } from "@uma/common";
+import { Coingecko, across } from "@uma/sdk";
+import { getAddress } from "@uma/contracts-node";
+
+import { RelaySubmitType } from "./Relayer";
+
+import type { BN } from "@uma/common";
+
+export enum TokenType {
+  WETH,
+  ERC20,
+  UMA,
+}
+
+export class ProfitabilityCalculator {
+  public l1TokenInfo: { [token: string]: { tokenType: TokenType; tokenEthPrice: BN } } = {};
+
+  public relayerDiscount: BN;
+
+  private readonly coingecko;
+  /**
+   * @notice Constructs new Profitability Calculator Instance.
+   * @param {Object} logger Module used to send logs.
+   * @param {Array }l1Tokens list of L1 tokens the relayer is running on. Profitability module will store the prices of
+   * these tokens and use them when computing if a relay should be executed or not.
+   * @param {Number} l1ChainId L1 chain id that the relayer is running on.
+   * @param {Number} relayerDiscount Relayer discount applied to relays to offset gas cost.
+   */
+  constructor(
+    readonly logger: winston.Logger,
+    readonly l1Tokens: string[],
+    readonly l1ChainId: number,
+    readonly relayerDiscountNumber: number = 0
+  ) {
+    this.relayerDiscount = toBNWei(relayerDiscountNumber).div(toBN("100"));
```

Comment:
Will this work? I thought BNs couldn't represent decimals?

Reply:
this is not storying a decimal. it is storing a Wei'd BN. `relayerDiscountNumber` is a number between 0 and 100. this is numerically taking relayerDiscountNumber*1e18/100. so if `relayerDiscountNumber` was 50 (for 50%) this would generate 50*1e18/100= 5E17 to represent 50%. 

you are right though that `relayerDiscountNumber` could be a fraction would cause issues on this. I have updated it to round down to the nearest number to avoid this.

Reply:
Ah I missed the Wei part of the BN conversion.